src/Coalesce.Framework.Persister.Elasticsearch/src/main/java/com/incadencecorp/coalesce/framework/persistance/elasticsearch/ElasticSearchIterator.java

```diff
@@ -228,16 +228,21 @@ private DocWriteRequest visitObject(CoalesceObject object, String index, String
 
         for (CoalesceField field : record.getFields())
         {
-            if (field.isFlatten() && field.getBaseValue() != null)
+            if (field.isFlatten() && (field.getBaseValue() != null || field.getDataType() == ECoalesceFieldDataTypes.FILE_TYPE))
```

Comment:
Can the field getBaseValue be null for FILE_TYPE?
Should the file also be checked for empty before being set?

Reply:
The base value will almost always be null. This comes from the .NET days where files were being stored as base64 within the base value. This clearly would not scale so we use the field's key as the unique filename (to avoid conflicts) and store the original file's name along with the hash in the extended attributes. However there still remains a flag which is off by default to persist the file's content as base64 within the XML as the base value. We also created a BINARY_TYPE so if you have a field that will only contain small (few hundred bytes) files and wanted to store it within the XML you can without storing all, potentially large, file types. So the simple answer to your question is; its treated different because the file's name is stored in the extended attributes and not the field's value.